domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/stops/BoardingArea.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2020. MobilityData IO.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.domain.entity.stops;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfs.EntityBuildResult;
+import org.mobilitydata.gtfsvalidator.domain.entity.notice.error.MissingRequiredValueNotice;
+
+import java.util.List;
+
+/**
+ * Model class for an entity defined in stops.txt with location_type = 4
+ */
+public class BoardingArea extends LocationBase {
+
+    @NotNull
+    private final String parentStation;
+
+    private BoardingArea(@NotNull final String stopId,
+                         @Nullable final String stopCode,
+                         @Nullable final String stopName,
+                         @Nullable final String stopDesc,
+                         @Nullable final Float stopLat,
+                         @Nullable final Float stopLon,
+                         @Nullable final String zoneId,
+                         @Nullable final String stopUrl,
+                         @NotNull final String parentStation,
+                         @Nullable final String stopTimezone,
+                         @Nullable final List<String> children) {
+        super(stopId, stopCode, stopName, stopDesc, stopLat, stopLon, zoneId, stopUrl, stopTimezone, children);
+        this.parentStation = parentStation;
+    }
+
+    public @NotNull String getParentStation() {
+        return parentStation;
+    }
+
+    public static class BoardingAreaBuilder extends LocationBaseBuilder {
+
+        private String parentStation;
+
+        public BoardingAreaBuilder parentStation(final String parentStation) {
+            this.parentStation = parentStation;
+            return this;
+        }
+
+        public EntityBuildResult<?> build() {
```

Comment:
If I'm reading the spec right, stop `location_types` of `2`, `3`, and `4` (entrances, nodes, and boarding areas) shouldn't have children stops (i.e., they must be the leaf nodes). Should we be checking for a non-null children list here (and in the other relevant builders) and returning this as a notice?

Reply:
Oh right could point. Will change

Reply:
Thought more about that while implementing. Because we validate parent `location_type` for any stop with a `parent_station`, that should be equivalent. Also, because parent --> child is not modeled directly in the dataset, the resulting notice would run the risk of being hard to decipher by users of the validator. For those reason, I think checking children in the builder is not nescessary.